### PR TITLE
Start pod addressability reconciler with the controller

### PR DESCRIPTION
**Pull request by markusthoemmes**
# Changes

Ref https://github.com/knative/serving/issues/10751

This enables the new reconciler and adjusts the necessary ClusterRole. Note that the reconciler is disabled by default, i.e. it'll do nothing.

/assign @nak3 @julz @ZhiminXiang @vagababov

### CODE REVIEW

**Review by julz — DISMISSED** [review]
lgtm

**Review by nak3 — APPROVED** [review]
/lgtm

mesh test failed but it seems unrelated issue.

File: test/config/300-serverlessservice.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Copyright 2019 The Knative Authors
```
Comment (vagababov) [inline test/config/300-serverlessservice.yaml:1]: 2021.
Is there anything we need to keep in sync?

Comment (markusthoemmes) [reply test/config/300-serverlessservice.yaml:1]: Already sent the respective PRs to clean up the copied CRDs.

**Pull request merged**